File: game/Assets/Scripts/Network/Messages/BugReport.cs
using System;
using System.Collections.Generic;

namespace Network
{
    [Serializable]
    public class ModuleLog
    {
        public string module;
        public string log;
    }

    [Serializable]
    public class BugReport
    {
        public MapUpdate map_update;
        public List<TurnState> turn_state_log;
        public StateSync state_sync;
        public List<ModuleLog> logs;
    }
}

File: src/cb2game/server/messages/bug_report.py
from dataclasses import dataclass, field
from typing import List, Optional

from mashumaro.mixins.json import DataClassJSONMixin

from cb2game.server.messages.map_update import MapUpdate
from cb2game.server.messages.state_sync import StateSync
from cb2game.server.messages.turn_state import TurnState


@dataclass(frozen=True)
class ModuleLog(DataClassJSONMixin):
    module: str
    log: str


@dataclass(frozen=True)
class BugReport(DataClassJSONMixin):
    logs: List[ModuleLog]
    turn_state_log: List[TurnState] = field(default_factory=list)
    state_sync: Optional[StateSync] = None
    map_update: Optional[MapUpdate] = None
